spaceapi/src/lib.rs

```diff
@@ -8,11 +8,35 @@ use rustc_serialize::json::{Json, ToJson};
 /// An Optional value can contain Optional::Some<T> or Optional::Absent.
 /// It is similar to an Option, but Optional::Absent means it will be
 /// omitted when serialized.
+#[derive(Debug, Copy, Clone)]
 pub enum Optional<T> {
     Value(T),
     Absent,
 }
 
+impl<T> Optional<T> {
+    pub fn map_or<U, F: FnOnce(T) -> U>(self, def: U, f: F) -> U {
+        match self {
+            Optional::Value(v) => f(v),
+            Optional::Absent => def,
+        }
+    }
+
+    pub fn as_mut<'r>(&'r mut self) -> Optional<&'r mut T> {
+        match *self {
+            Optional::Value(ref mut x) => Optional::Value(x),
+            Optional::Absent => Optional::Absent
+        }
+    }
```

Comment:
You mean ``as_mut``? It's the complementary to ``as_ref`` that's why I added it. But it's currently not used.